internal/painter/gl/draw.go

```diff
@@ -216,9 +217,44 @@ func (p *painter) drawText(text *canvas.Text, pos fyne.Position, frame fyne.Size
 		pos = fyne.NewPos(pos.X, pos.Y+(containerSize.Height-size.Height)/2)
 	}
 
+	color := text.Color
+	if color == nil {
+		color = theme.ForegroundColor()
+	}
+
 	// text size is sensitive to position on screen
 	size, _ = roundToPixelCoords(size, text.Position(), p.pixScale)
-	p.drawTextureWithDetails(text, p.newGlTextTexture, pos, size, frame, canvas.ImageFillStretch, 1.0, 0)
+	p.drawGrayTexture(text, p.newGlTextTexture, pos, size, frame, color, 0)
+}
+
+func (p *painter) drawGrayTexture(o fyne.CanvasObject, creator func(canvasObject fyne.CanvasObject) Texture,
```

Comment:
Passing in `Color` to something called `Gray` seems a little misleading. Should it be more like `drawUniformTexture` or `DrawSingleChannelTexture`?